data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java

```diff
@@ -198,4 +255,197 @@ public void geEnvironment() throws StorageException, EnvironmentNotFoundExceptio
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(thirdRecordSameCluster);
+
+    when(environmentTargetVersionDDBRecordPaginatedScanList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenReturn(environmentTargetVersionDDBRecordPaginatedScanList);
+
+    final List<String> clusters = environmentRepositoryDDB.listClusters();
+
+    final Set<String> expectedClusters = new HashSet<>();
+    for (EnvironmentTargetVersionDDBRecord record : environmentTargetVersionDDBRecords) {
+      expectedClusters.add(record.getCluster());
+    }
+
+    assertEquals(expectedClusters.size(), clusters.size());
+    for (String c : expectedClusters) {
+      if (!clusters.contains(c)) {
+        fail(String.format("Does not contain %s", c));
+      }
+    }
+
+    verify(dynamoDBMapper)
+        .scan(eq(EnvironmentTargetVersionDDBRecord.class), dynamoDBScanExpressionCaptor.capture());
+    final DynamoDBScanExpression expectedDynamodbScanExpression =
+        new DynamoDBScanExpression()
+            .withIndexName(EnvironmentTargetVersionDDBRecord.ENVIRONMENT_CLUSTER_GSI_NAME)
+            .withConsistentRead(false);
+
+    assertEquals(
+        expectedDynamodbScanExpression.getIndexName(),
+        dynamoDBScanExpressionCaptor.getValue().getIndexName());
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void listEnvironmentIdsByClusterNullCluster() throws StorageException {
+    environmentRepositoryDDB.listEnvironmentIdsByCluster(null);
+  }
+
+  @Test(expected = StorageException.class)
+  public void listEnvironmentIdsByClusterStorageException() throws StorageException {
+    when(dynamoDBMapper.query(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBQueryExpression.class)))
+        .thenThrow(new AmazonServiceException(""));
+    environmentRepositoryDDB.listEnvironmentIdsByCluster(CLUSTER);
+  }
+
+  @Test
+  public void listEnvironmentIdsByCluster() throws StorageException {
+    List<EnvironmentTargetVersionDDBRecord> environmentTargetVersionDDBRecords = new ArrayList<>();
+    environmentTargetVersionDDBRecords.add(environmentTargetVersionDDBRecord);
+    final EnvironmentTargetVersionDDBRecord secondRecord =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(environmentTargetVersionDDBRecord.getCluster())
+            .build();
+    environmentTargetVersionDDBRecords.add(secondRecord);
+    when(environmentTargetVersionDDBRecordPaginatedQueryList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.query(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBQueryExpression.class)))
```

Comment:
The `isA(DynamoDBQueryExpression.class)` causes an `unchecked assignment` warning here because of the raw type. Since we're already testing the actual value captured in a test case below, perhaps just using `any()` here is sufficient?